src/app/core/tooltip/tooltip.service.ts

```diff
@@ -0,0 +1,42 @@
+import { ElementRef, Injectable } from "@angular/core";
+import {
+  Overlay,
+  OverlayPositionBuilder,
+  OverlayRef,
+} from "@angular/cdk/overlay";
+import { ComponentPortal } from "@angular/cdk/portal";
+import { TooltipComponent } from "./tooltip/tooltip.component";
+
+@Injectable({
+  providedIn: "root",
+})
+export class TooltipService {
+  constructor(
+    private overlay: Overlay,
+    private overlayPositionBuilder: OverlayPositionBuilder
+  ) {}
+
+  public createTooltip(elementRef: ElementRef): OverlayRef {
```

Comment:
I think we shouldn't be building our own tooltip module as there is MatTooltip available, which we are already using in other places. Having different implementations will inevitable lead to confusion and inconsistent styling.

Do you think it is possible to use MatTooltipModule to cover your logic from this? Or maybe we need to implement a permissionProtectedButton component instead of a directive, so we can apply styling and matTooltip on that?